packages/babel-core/src/config/partial.js

```diff
@@ -81,22 +81,29 @@ export default function* loadPrivatePartialConfig(
     root: rootDir = ".",
     rootMode = "root",
     caller,
+    showConfig = false,
   } = args;
   const absoluteCwd = path.resolve(cwd);
   const absoluteRootDir = yield* resolveRootMode(
     path.resolve(absoluteCwd, rootDir),
     rootMode,
   );
 
+  const filename =
+    typeof args.filename === "string"
+      ? path.resolve(cwd, args.filename)
+      : undefined;
+
   const context: ConfigContext = {
-    filename:
-      typeof args.filename === "string"
-        ? path.resolve(cwd, args.filename)
-        : undefined,
+    filename,
     cwd: absoluteCwd,
     root: absoluteRootDir,
     envName,
     caller,
+    showConfig:
+      showConfig ||
+      path.resolve(cwd, process.env.BABEL_SHOW_CONFIG_FOR_PATH ?? "") ===
+        filename,
```

Comment:
Maybe we should also allow setting `showConfig` to a string? I'm trying to think about one usecase where having a boolean and logging the config for _every_ file is useful, but I can't think of any common one :thinking:

Reply:
I am not sure if logging the config for _every file_ is useful, because it can overwhelm quite easily for even a small project.

When people is not using `@babel/cli`, I prefer `env.BABEL_SHOW_CONFIG_FOR_PATH` over `showConfig: boolean` because environment variable is more robust, since people may end up adding `showConfig` to a babel config that is, unfortunately, _not_ loaded by `@babel/core` at all, and I definitely don't want people debugging `showConfig` itself.

Reply:
> I am not sure if logging the config for every file is useful, because it can overwhelm quite easily for even a small project.

Than maybe we don't need the boolean `showConfig`, and only rely on the env var?

> Do you think we should throw if path.resolve(cwd, process.env.BABEL_SHOW_CONFIG_FOR_PATH ?? "") does not even exist? This is the case when people specify it wrong and currently no config will be printed.

:+1:

Reply:
I have removed `showConfig: boolean` support to make this PR minimal. We can add `showConfig: boolean` later if it is indeed a popular usage.